### PR TITLE
REF: Make unit part of address array

### PR DESCRIPTION
**Pull request by marcosrdz**
## Summary by CodeRabbit

- **New Features**
	- Enhanced payment destination management by introducing optional Bitcoin unit specification for each address
	- Improved address processing and transaction creation with more flexible unit handling

- **Refactor**
	- Streamlined state management for Bitcoin transaction units
	- Simplified fee selection and address unit tracking

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

File: screen/send/SendDetails.tsx [hunk 1]
```diff
@@ -110,8 +112,10 @@ const SendDetails = () => {
   const [dumb, setDumb] = useState(false);
   const { isEditable } = routeParams;
   // if utxo is limited we use it to calculate available balance
-  const balance: number = utxos ? utxos.reduce((prev, curr) => prev + curr.value, 0) : (wallet?.getBalance() ?? 0);
-  const allBalance = formatBalanceWithoutSuffix(balance, BitcoinUnit.BTC, true);
+  const balance = useMemo(() => {
+    return utxos ? utxos.reduce((prev, curr) => prev + curr.value, 0) : (wallet?.getBalance() ?? 0);
+  }, [utxos, wallet]);
```
Comment (limpbrains) [inline screen/send/SendDetails.tsx:117]: You can't use Memo here
Wallet will never change even if it's balance does. You will never get new balance

**Review by limpbrains — CHANGES_REQUESTED** [review]

**Comment by limpbrains** [comment]
A little bit of explanation would be nice to understand on what are fixing here

File: screen/send/SendDetails.tsx (hunk 1 above)
Comment (marcosrdz) [reply screen/send/SendDetails.tsx]: Reverted

**Comment by marcosrdz** [comment]
> A little bit of explanation would be nice to understand on what are fixing here 

To deal with one less useState

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**